packages/nimble-components/src/table/tests/table.spec.ts

```diff
@@ -1,7 +1,171 @@
+import { html } from '@microsoft/fast-element';
 import { Table } from '..';
+import { waitForUpdatesAsync } from '../../testing/async-helpers';
+import { type Fixture, fixture } from '../../utilities/tests/fixture';
+import type { TableData } from '../types';
+import { TablePageObject } from './table.pageobject';
+
+interface FakeTableData extends TableData {
+    stringData: string;
+    numericData: number;
+    booleanData: boolean;
+    dateData: Date;
+}
+const fakeTableDataKeys = [
+    'stringData',
+    'numericData',
+    'booleanData',
+    'dateData'
+];
+
+async function setup(): Promise<Fixture<Table<FakeTableData>>> {
+    return fixture<Table<FakeTableData>>(html`<nimble-table></nimble-table>`);
+}
+
+function makeFakeData(count: number): FakeTableData[] {
+    const data: FakeTableData[] = [];
+    for (let i = 0; i < count; i++) {
+        data.push({
+            stringData: `string #${i}`,
+            numericData: i * 2,
+            booleanData: i % 2 === 0,
+            // Generate arbitrary dates that are 10 seconds apart
+            dateData: new Date(1670000000000 + i * 10000)
+        });
+    }
+    return data;
+}
 
 describe('Table', () => {
+    let element: Table<FakeTableData>;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: TablePageObject<FakeTableData>;
+
+    function verifyRenderedData(expectedData: FakeTableData[]): void {
+        const expectedRowCount = expectedData.length;
+        expect(pageObject.getRenderedRowCount()).toEqual(expectedRowCount);
+        for (let rowIndex = 0; rowIndex < expectedRowCount; rowIndex++) {
+            for (
+                let columnIndex = 0;
+                columnIndex < fakeTableDataKeys.length;
+                columnIndex++
+            ) {
+                const dataKey = fakeTableDataKeys[columnIndex]!;
+                const expectedCellData = expectedData[rowIndex]![dataKey]!;
+                expect(
+                    pageObject.getCellContent(rowIndex, columnIndex)
+                ).toEqual(expectedCellData.toString());
+            }
+        }
+    }
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        pageObject = new TablePageObject<FakeTableData>(element);
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
     it('can construct an element instance', () => {
         expect(document.createElement('nimble-table')).toBeInstanceOf(Table);
     });
+
+    it('should render column headers', async () => {
+        await connect();
+
+        const fakeData = makeFakeData(1);
+        element.data = fakeData;
+        await waitForUpdatesAsync();
+
```

Comment:
Should we also assert that the number of column headers matches the number of fake table data keys?